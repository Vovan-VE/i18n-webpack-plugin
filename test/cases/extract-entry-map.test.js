import processFile from '../cases.setup';

describe('extract-entry-map', () => {
  const extracted = {};

  beforeAll(() => processFile([
    'extract-entry-map-bar.code.js',
    'extract-entry-map-foo.code.js',
  ])
    .then(({ files }) => {
      Object.keys(files).forEach((name) => {
        extracted[name] = require.requireActual(files[name]);
      });
    }));

  it('should return extracted keys', () => {
    expect(extracted).toMatchSnapshot();
  });
});
