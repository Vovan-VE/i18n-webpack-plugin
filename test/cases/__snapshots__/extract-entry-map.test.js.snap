// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extract-entry-map should return extracted keys 1`] = `
Object {
  "extract-entry-map-bar.tmp.json": Object {
    "category2": Object {
      "Message 3": "",
    },
    "category3": Object {
      "Message 4": "",
    },
    "category4": Object {
      "Message 5": "",
    },
  },
  "extract-entry-map-foo.tmp.json": Object {
    "category1": Object {
      "Message 1": "",
      "Message 2": "",
    },
    "category2": Object {
      "Message 3": "",
      "Message imported": "",
    },
  },
}
`;
